hudi-hadoop-common/src/main/java/org/apache/hudi/hadoop/fs/HadoopFSUtils.java

```diff
@@ -154,4 +158,90 @@ public static FileStatus convertToHadoopFileStatus(StoragePathInfo pathInfo) {
         pathInfo.getModificationTime(),
         convertToHadoopPath(pathInfo.getPath()));
   }
+
+  /**
+   * Fetch the right {@link FSDataInputStream} to be used by wrapping with required input streams.
+   *
+   * @param fs         instance of {@link FileSystem} in use.
+   * @param filePath   path of the file.
+   * @param bufferSize buffer size to be used.
+   * @return the right {@link FSDataInputStream} as required.
+   */
+  public static FSDataInputStream getFSDataInputStream(FileSystem fs,
+                                                       StoragePath filePath,
+                                                       int bufferSize) {
+    FSDataInputStream fsDataInputStream = null;
+    try {
+      fsDataInputStream = fs.open(new Path(filePath.toUri()), bufferSize);
```

Comment:
What do you think about a helper method for this?:
new Path(filePath.toUri())

or will that just add more clutter?

Reply:
Good catch.  We have the util method.  Fixed.